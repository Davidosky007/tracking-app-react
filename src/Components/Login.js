import { Redirect, useLocation } from 'react-router-dom';
import { useState } from 'react';
import { login } from '../logic/sessions';
import { saveToken } from '../logic/storage';

const Login = () => {
  const [redirect, setRedirect] = useState();
  const [error, setError] = useState();

  const location = useLocation();
  const path = location.pathname.split('/')[2];

  const signUp = path !== 'login';
  const submitValue = signUp ? 'Sign Up' : 'Login';
  const endpoint = signUp ? 'users' : 'session';
  const validatePassword = signUp ? <input className="field m-b-20 background-blue color-white" id="rPassword" type="password" placeholder="reapeat password" /> : <div />;
  const link = signUp ? <a href="/session/login">Login</a> : <a href="/users/sign-up">Sign Up</a>;

  const handleSubmit = async (e) => {
    e.preventDefault();
    const name = e.target.parentElement[0].value;
    const password = e.target.parentElement[1].value;
    const rPassword = e.target.parentElement[2].value;
    if (signUp && password !== rPassword) return;
    const response = await login(name, password, endpoint);
    if (response.token) {
      saveToken(response.token);
      setRedirect(<Redirect to={{ pathname: '/' }} />);
    } else {
      setError(<div className="m-b-20">{response.error}</div>);
    }
  };

  return (
    <div className="login text-center">
      {redirect}
      {error}
      <form>
        <input className="field m-b-20 background-blue color-white" id="name" type="text" placeholder="username" />
        <input className="field m-b-20 background-blue color-white" id="password" type="password" placeholder="password" />
        {validatePassword}
        <input className="login-btn background-gray m-b-30" type="submit" value={submitValue} onClick={(e) => handleSubmit(e)} />
      </form>
      <p>or </p>
      {link}
    </div>
  );
};

export default Login;
