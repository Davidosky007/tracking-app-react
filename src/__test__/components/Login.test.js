import renderer from 'react-test-renderer';
import Login from '../../Components/Login';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: 'example.com/test',
  }),
}));

describe('Login', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Login />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
