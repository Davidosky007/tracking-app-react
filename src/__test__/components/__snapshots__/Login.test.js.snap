// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly 1`] = `
<div
  className="login text-center"
>
  <form>
    <input
      className="field m-b-20 background-blue color-white"
      id="name"
      placeholder="username"
      type="text"
    />
    <input
      className="field m-b-20 background-blue color-white"
      id="password"
      placeholder="password"
      type="password"
    />
    <input
      className="field m-b-20 background-blue color-white"
      id="rPassword"
      placeholder="reapeat password"
      type="password"
    />
    <input
      className="login-btn background-gray m-b-30"
      onClick={[Function]}
      type="submit"
      value="Sign Up"
    />
  </form>
  <p>
    or 
  </p>
  <a
    href="/users/session"
  >
    Login
  </a>
</div>
`;
